packages/builder/src/schemas.zod.ts

```diff
@@ -0,0 +1,304 @@
+import { z } from 'zod';
+import { handleZodErrors } from './error/zod';
+
+/// ================================ INPUT CONFIG SCHEMAS ================================ \\\
+
+export const contractSchema = z
+  .object({
+    artifact: z.string(),
+  })
+  .merge(
+    z
+      .object({
+        create2: z.boolean(),
+        from: z.string(),
+        nonce: z.string(),
+        abi: z.string(),
+        abiOf: z.array(z.string()),
+        args: z.array(z.any()),
+        libraries: z.record(z.string()),
+
+        // used to force new copy of a contract (not actually used)
+        salt: z.string(),
+
+        value: z.string(),
+        overrides: z.object({
+          gasLimit: z.string(),
+        }),
+
+        depends: z.array(z.string()),
```

Comment:
this shuold be validated to be `<step type>.<name>` format. can probably do regex match here at least.